pkg/event/dao/dao.go

```diff
@@ -120,4 +134,40 @@ func (d *dao) GetCursor(ctx context.Context) (*models.EventCursor, error) {
 	return &eventIndex, nil
 }
 
+func (d *dao) GetCursors(ctx context.Context, cursorType models.EventCursorType,
+	regionIDs ...uint) ([]*models.EventCursor, error) {
+	var eventIndex []*models.EventCursor
+	statement := d.db.WithContext(ctx).Where("type = ?", cursorType)
+	if len(regionIDs) > 0 {
+		statement.Where("region_id in (?)", regionIDs)
+	}
+	if result := statement.Find(&eventIndex); result.Error != nil {
+		if result.Error == gorm.ErrRecordNotFound {
+			return nil, herrors.NewErrNotFound(herrors.EventCursorInDB,
+				result.Error.Error())
+		}
+		return nil, herrors.NewErrGetFailed(herrors.EventCursorInDB, result.Error.Error())
+	}
+	return eventIndex, nil
+}
+
+func (d *dao) DeleteEvents(ctx context.Context, id ...uint) (int64, error) {
+	var events []*models.Event
+	tx := d.db.WithContext(ctx).Begin()
+
+	result := tx.Where("id in (?)", id).Delete(&events)
+	if result.Error != nil {
+		tx.Rollback()
+		return 0, herrors.NewErrDeleteFailed(herrors.EventInDB, result.Error.Error())
+	}
+
+	if result := tx.Where("event_id in (?)", id).Delete(&webhookmodels.WebhookLog{}); result.Error != nil {
+		tx.Rollback()
+		return 0, herrors.NewErrDeleteFailed(herrors.WebhookLogInDB, result.Error.Error())
+	}
+
+	tx.Commit()
+	return result.RowsAffected, nil
+}
```

Comment:
```suggestion
func (d *dao) DeleteEvents(ctx context.Context, id ...uint) (int64, error) {
	var events []*models.Event
	tx := d.db.WithContext(ctx).Begin()

	result := tx.Where("id in (?)", id).Delete(&events)
	if result.Error != nil {
		tx.Rollback()
		return 0, herrors.NewErrDeleteFailed(herrors.EventInDB, result.Error.Error())
	}

	if result := tx.Where("event_id in (?)", id).Delete(&webhookmodels.WebhookLog{}); result.Error != nil {
		tx.Rollback()
		return 0, herrors.NewErrDeleteFailed(herrors.WebhookLogInDB, result.Error.Error())
	}

	tx.Commit()
	return result.RowsAffected, nil
}
```
1. uint:
```go
if len(id) == 0 ?
```

2. What about the delete event timeout?

Reply:
we have no timeout when operating the db